diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index e758ab3b9265..695fbc71b2e5 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -313,11 +313,6 @@ set_define(toolkit_define, True)
 def toolkit_gtk(toolkit):
     return toolkit == "gtk"
 
-
-set_config("MOZ_X11", True, when=toolkit_gtk)
-set_define("MOZ_X11", True, when=toolkit_gtk)
-add_old_configure_assignment("MOZ_X11", True, when=toolkit_gtk)
-
 # Wayland support
 # ==============================================================
 wayland_headers = pkg_check_modules(
@@ -338,6 +333,20 @@ def wayland_headers(wayland, toolkit_gtk, artifacts):
 set_config("MOZ_WAYLAND", depends_if(wayland_headers)(lambda _: True))
 set_define("MOZ_WAYLAND", depends_if(wayland_headers)(lambda _: True))
 
+# X11 optional if wayland is enabled
+# ==============================================================
+
+x11_headers = pkg_check_modules(
+    "MOZ_X11",
+    "gtk+-x11-3.0 >= 3.14",
+    allow_missing=depends(full_toolkit)(lambda t: t == "cairo-gtk3-wayland"),
+    when=toolkit_gtk,
+)
+
+set_config("MOZ_X11", depends_if(x11_headers)(lambda _: True))
+set_define("MOZ_X11", depends_if(x11_headers)(lambda _: True))
+add_old_configure_assignment("MOZ_X11", depends_if(x11_headers)(lambda _: True))
+
 # GL Provider
 # ==============================================================
 option("--with-gl-provider", nargs=1, help="Set GL provider backend type")
