diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/BUILD.gn b/third_party/libwebrtc/webrtc/modules/desktop_capture/BUILD.gn
index d1033cbd71b0..e21849d96bda 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/BUILD.gn
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/BUILD.gn
@@ -356,6 +356,12 @@ rtc_static_library("desktop_capture_generic") {
     }
   }
 
+  if (rtc_use_pipewire && !use_x11) {
+    sources += [
+      "desktop_device_info_null.cc",
+    ]
+  }
+
   if (rtc_use_pipewire) {
     sources += [
       "linux/base_capturer_pipewire.cc",
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_capturer.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_capturer.cc
index 5e4dcb058d16..c40297d9073b 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_capturer.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_capturer.cc
@@ -15,8 +15,10 @@
 
 #if defined(WEBRTC_USE_PIPEWIRE) || defined(USE_X11)
 #include <gtk/gtk.h>
+#if defined(USE_X11)
 #include <gtk/gtkx.h>
 #endif
+#endif
 
 namespace webrtc {
 
diff --git a/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_device_info_null.cc b/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_device_info_null.cc
index 20a35689281b..0229f053337a 100644
--- a/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_device_info_null.cc
+++ b/third_party/libwebrtc/webrtc/modules/desktop_capture/desktop_device_info_null.cc
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#include "webrtc/modules/desktop_capture/desktop_device_info.h"
+#include "modules/desktop_capture/desktop_device_info.h"
 
 namespace webrtc {
 
@@ -40,4 +40,8 @@ void
 DesktopDeviceInfoNull::InitializeApplicationList() {
 }
 
+void
+DesktopDeviceInfoNull::InitializeTabList() {
+}
+
 } //namespace webrtc
