diff -uNr freeciv-2.6.2.orig/client/gui-qt/canvas.cpp freeciv-2.6.2/client/gui-qt/canvas.cpp
--- freeciv-2.6.2.orig/client/gui-qt/canvas.cpp	2020-02-15 09:49:02.000000000 -0600
+++ freeciv-2.6.2/client/gui-qt/canvas.cpp	2020-11-14 21:14:40.447691916 -0600
@@ -18,6 +18,7 @@
 // Qt
 #include <QFontMetrics>
 #include <QPainter>
+#include <QPainterPath>
 
 // qt-client
 #include "canvas.h"
@@ -309,7 +310,7 @@
   afont = get_font(font);
   fm = new QFontMetrics(*afont);
   if (width) {
-    *width = fm->width(QString::fromUtf8(text));
+    *width = fm->horizontalAdvance(QString::fromUtf8(text));
   }
 
   if (height) {
diff -uNr freeciv-2.6.2.orig/client/gui-qt/citydlg.cpp freeciv-2.6.2/client/gui-qt/citydlg.cpp
--- freeciv-2.6.2.orig/client/gui-qt/citydlg.cpp	2020-02-15 09:49:02.000000000 -0600
+++ freeciv-2.6.2/client/gui-qt/citydlg.cpp	2020-11-14 21:39:24.227711683 -0600
@@ -27,6 +27,7 @@
 #include <QPainter>
 #include <QRadioButton>
 #include <QRect>
+#include <QScreen>
 #include <QScrollArea>
 #include <QScrollBar>
 #include <QSplitter>
diff -uNr freeciv-2.6.2.orig/client/gui-qt/dialogs.cpp freeciv-2.6.2/client/gui-qt/dialogs.cpp
--- freeciv-2.6.2.orig/client/gui-qt/dialogs.cpp	2020-02-15 09:49:02.000000000 -0600
+++ freeciv-2.6.2/client/gui-qt/dialogs.cpp	2020-11-14 20:39:17.900663639 -0600
@@ -24,6 +24,7 @@
 #include <QMessageBox>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QPainterPath>
 #include <QRadioButton>
 #include <QRect>
 #include <QSignalMapper>
diff -uNr freeciv-2.6.2.orig/client/gui-qt/ratesdlg.cpp freeciv-2.6.2/client/gui-qt/ratesdlg.cpp
--- freeciv-2.6.2.orig/client/gui-qt/ratesdlg.cpp	2020-02-15 09:49:02.000000000 -0600
+++ freeciv-2.6.2/client/gui-qt/ratesdlg.cpp	2020-11-14 22:10:21.609736427 -0600
@@ -21,6 +21,7 @@
 #include <QGroupBox>
 #include <QMouseEvent>
 #include <QPainter>
+#include <QScreen>
 #include <QVBoxLayout>
 
 // common
@@ -128,8 +129,7 @@
   Multipler rates dialog constructor
   Inheriting from qfc_dialog will cause crash in Qt5.2
 **************************************************************************/
-multipler_rates_dialog::multipler_rates_dialog(QWidget *parent,
-                                               Qt::WindowFlags f)
+multipler_rates_dialog::multipler_rates_dialog(QWidget *parent)
   : QDialog(parent)
 {
   QGroupBox *group_box;
diff -uNr freeciv-2.6.2.orig/client/gui-qt/ratesdlg.h freeciv-2.6.2/client/gui-qt/ratesdlg.h
--- freeciv-2.6.2.orig/client/gui-qt/ratesdlg.h	2020-02-15 09:49:02.000000000 -0600
+++ freeciv-2.6.2/client/gui-qt/ratesdlg.h	2020-11-14 21:25:46.707700792 -0600
@@ -99,7 +99,7 @@
   Q_OBJECT
 
 public:
-  explicit multipler_rates_dialog(QWidget* parent = 0, Qt::WindowFlags f = 0);
+  explicit multipler_rates_dialog(QWidget* parent = 0);
 private:
   QList<QSlider*> slider_list;
   QPushButton *cancel_button;
